src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Web/Debugger/WebLaunchSettingsProvider.cs

```diff
@@ -0,0 +1,301 @@
+            public const string IISExpressSSLPort = nameof(IISExpressSSLPort);
+            public const string IISExpressUseClassicPipelineMode = nameof(IISExpressUseClassicPipelineMode);
+            public const string UseGlobalApplicationHostFile = nameof(UseGlobalApplicationHostFile);
+        }
+
+        private const string WAPProjectGuidString = "{" + ProjectType.LegacyWebApplicationProject + "}";
+        private readonly IUnconfiguredProjectCommonServices _projectServices;
+        private readonly IProjectExtensionDataService _projectExtensionDataService;
+        private FlavorServerSettings? _flavorServerSettings;
+
+        private bool _useIISExpress;
+        private bool _use64BitIISExpress;
+        private int _issExpressSSLPort;
+        private bool _useWindowsAuth;
+        private bool _useAnonymousAuth;
+        private bool _useClassicPipelineMode;
+        private bool _useGlobalApplicationHostFile;
+
+        [ImportingConstructor]
+        public WebLaunchSettingsProvider(
+            ILaunchSettingsProvider2 launchSettingsProvider,
+            IUnconfiguredProjectCommonServices projectServices,
+            IProjectExtensionDataService projectExtensionDataService,
+            IProjectThreadingService threadingService)
+             : base(threadingService.JoinableTaskContext)
+        {
+            _projectServices = projectServices;
+            _projectExtensionDataService = projectExtensionDataService;
+        }
+
+        public async Task<WebLaunchSettings> GetLaunchSettingsAsync()
+        {
+            await InitializeAsync();
+
+            List<string> urls = new List<string>();
+            ServerType serverType = ServerType.IISExpress;
+
+            if (_flavorServerSettings == null)
+            {
+                return new WebLaunchSettings(serverType, urls, _useWindowsAuth, _useAnonymousAuth, _useClassicPipelineMode, _use64BitIISExpress, _useGlobalApplicationHostFile, false, null);
+            }
+
+            if (_flavorServerSettings.UseCustomServer)
+            {
+                if (_flavorServerSettings.CustomServerUrl != null)
+                {
+                    urls.Add(_flavorServerSettings.CustomServerUrl);
+                }
+
+                serverType = ServerType.Custom;
+            }
+            else if (_flavorServerSettings.UseIIS)
+            {
+                if (_useIISExpress)
+                {
+                    if (_flavorServerSettings.IISUrl != null)
+                    {
+                        urls.Add(_flavorServerSettings.IISUrl);
+                        if (_issExpressSSLPort != 0 && !_flavorServerSettings.IISUrl.StartsWith("https://", StringComparison.OrdinalIgnoreCase))
+                        {
+                            var uriBuilder = new UriBuilder(_flavorServerSettings.IISUrl);
+                            uriBuilder.Port = _issExpressSSLPort;
+                            uriBuilder.Scheme = "https://";
```

Comment:
```suggestion
                            uriBuilder.Scheme = Uri.UriSchemeHttps;
```

Could use an object initialiser for `uriBuilder`.